crml/cennzx-spot/src/lib.rs

```diff
@@ -961,4 +943,66 @@ impl<T: Trait> Module<T> {
 
 		Ok(())
 	}
+
+	/// Calculate the buy price of some asset for another
+	/// In simple terms: 'If I want to buy _x_ amount of asset _a_ how much of asset _b_ will it cost?'
+	/// `asset_to_buy` is the asset to buy
+	/// `amount_to_buy` is the amount of `asset_to_buy` required
+	/// `asset_to_pay` is the asset to use for payment (the final price will be given in this asset)
+	pub fn calculate_buy_price(
+		asset_to_buy: T::AssetId,
+		amount_to_buy: T::Balance,
+		asset_to_pay: T::AssetId,
+	) -> Result<T::Balance, DispatchError> {
+		ensure!(asset_to_buy != asset_to_pay, Error::<T>::AssetCannotSwapForItself);
+
+		// Find the cost of `amount_to_buy` of `asset_to_buy` in terms of core asset
+		// (how much core asset does it cost?).
+		let core_asset_amount = if asset_to_buy == Self::core_asset_id() {
+			amount_to_buy
+		} else {
+			Self::get_core_to_asset_output_price(&asset_to_buy, amount_to_buy, Self::fee_rate())?
+		};
+
+		// Find the price of `core_asset_amount` in terms of `asset_to_pay`
+		// (how much `asset_to_pay` does `core_asset_amount` cost?)
+		let pay_asset_amount = if asset_to_pay == Self::core_asset_id() {
+			core_asset_amount
+		} else {
+			Self::get_asset_to_core_output_price(&asset_to_pay, core_asset_amount, Self::fee_rate())?
+		};
+
+		Ok(pay_asset_amount)
+	}
+
+	/// Calculate the sell price of some asset for another
+	/// In simple terms: 'If I sell _x_ amount of asset _y_ how much of asset _z_ will I get in return?'
```

Comment:
✋  I used a,b here but y,z above, best to update to one or other for consistency